integrations.go

```diff
@@ -260,3 +260,104 @@ func getIgnoreErrorsSuspects(event *Event) []string {
 
 	return suspects
 }
+
+// ================================
+// Contextify Frames Integration
+// ================================
+
+type contextifyFramesIntegration struct {
+	sr              sourceReader
+	contextLines    int
+	cachedLocations map[string]string
+}
+
+func (cfi *contextifyFramesIntegration) Name() string {
+	return "ContextifyFrames"
+}
+
+func (cfi *contextifyFramesIntegration) SetupOnce(client *Client) {
+	cfi.sr = newSourceReader()
+	cfi.contextLines = 5
+	cfi.cachedLocations = make(map[string]string)
+
+	client.AddEventProcessor(cfi.processor)
+}
+
+func (cfi *contextifyFramesIntegration) processor(event *Event, hint *EventHint) *Event {
+	// Range over all exceptions
+	for _, ex := range event.Exception {
+		// If it has no stacktrace, just bail out
+		if ex.Stacktrace == nil {
+			continue
+		}
+
+		// If it does, it should have frames, so try to contextify them
+		ex.Stacktrace.Frames = cfi.contextify(ex.Stacktrace.Frames)
+	}
+
+	return event
+}
+
+func (cfi *contextifyFramesIntegration) contextify(frames []Frame) []Frame {
+	contextifiedFrames := make([]Frame, 0, len(frames))
+
+	for _, frame := range frames {
+		if !frame.InApp {
+			contextifiedFrames = append(contextifiedFrames, frame)
+			continue
+		}
+
+		var path string
+		// Optimize for happy path here
+		if fileExists(frame.AbsPath) {
+			path = frame.AbsPath
+		} else {
+			path = cfi.findNearbySourceCodeLocation(frame.AbsPath)
+		}
+
+		if path == "" {
+			contextifiedFrames = append(contextifiedFrames, frame)
+			continue
+		}
+
+		lines, initial := cfi.sr.readContextLines(path, frame.Lineno, cfi.contextLines)
```

Comment:
Blocking I/O?

Reply:
Yes, but it's all cached in the source reader once done

Reply:
When it comes to blocking I/O, there's no `yes, but` 👿 
I guess ASP.NET Core is a bad influence?! or is it?